07.ls_object/long_list_formatter.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module LongListFormatter
+  def long_format
+    max_length = Data.define(:nlink, :uid, :gid, :file_size, :file_name).new(* calc_max_length)
```

Comment:
```suggestion
    max_length = Data.define(:nlink, :uid, :gid, :file_size, :file_name).new(*calc_max_length)
```

スペースを空けない方が多いかも。

あと、ここの`:nlink, :uid, :gid, :file_size, :file_name`と、50行目の`[nlink, uid, gid, file_size, file_name]`をリンクさせる（カラムの順番を一致させる）のは「もろい」プログラムになる原因になります。

https://bootcamp.fjord.jp/pages/three-legged-race#-3

```suggestion
    max_length = hogehoge
```

のようにオブジェクトの作成そのものをメソッド化した方が堅牢になりそうです。

Reply:
`calc_max_length`でオブジェクトの作成までするようにします。

`*`はスペースを空けないようにします。